libfreerdp/core/settings.c

```diff
@@ -277,6 +277,199 @@ void freerdp_settings_print_warnings(const rdpSettings* settings)
 	}
 }
 
+static BOOL monitor_operlaps(const rdpSettings* settings, UINT32 start, UINT32 count,
+                             const rdpMonitor* compare)
+{
+	const RECTANGLE_16 rect1 = { .left = compare->x,
+		                         .top = compare->y,
+		                         .right = compare->x + compare->width,
+		                         .bottom = compare->y + compare->height };
+	for (UINT32 x = start; x < count; x++)
+	{
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+		const RECTANGLE_16 rect2 = { .left = monitor->x,
+			                         .top = monitor->y,
+			                         .right = monitor->x + monitor->width,
+			                         .bottom = monitor->y + monitor->height };
+
+		if (rectangles_intersects(&rect1, &rect2))
+		{
+			WLog_ERR(
+			    TAG,
+			    "Mulitimonitor mode requested, but local layout has gaps or overlapping areas!");
+			WLog_ERR(TAG, "Please reconfigure your local monitor setup so that thre are no gaps or "
+			              "overlapping areas!");
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
+
+static BOOL monitor_has_gaps(const rdpSettings* settings, UINT32 start, UINT32 count,
+                             const rdpMonitor* compare)
+{
+	BOOL hasNeighbor = FALSE;
+	for (UINT32 x = 0; x < count; x++)
```

Comment:
can we loop with `i` (or `j`) and not `x` as when quickly reading I though that was a variable for the x axis.